crates/circuit/src/dot_utils.rs

```diff
@@ -0,0 +1,109 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2022
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+// This module is forked from rustworkx at:
+// https://github.com/Qiskit/rustworkx/blob/c4256daf96fc3c08c392450ed33bc0987cdb15ff/src/dot_utils.rs
+// and has been modified to generate a dot file from a Rust DAGCircuit instead
+// of a rustworkx PyGraph object
```

Comment:
The license headers are different between the projects, rustworkx doesn't typically bother with the copyright line because that can be found from the git log, and it uses the standard header format recommended by apache. While in qiskit we have requirements on the header format to include "This code is part of Qiskit" and also the copyright line.

I normally would have put the copyright header from rustworkx directly below our copyright header to indicate the license of the original source, but because they're so similar it just looked like we copied the apache header twice. You're right though I can just include a statement that rustworkx is also licensed as apache 2.0 in my disclaimer about it being forked which I think should be sufficient here.

Reply:
For Rustworkx's copyright, I was more meaning that the owner of it isn't asserted in the Python sdist or wheels, both of which distribute its source.  I don't think it needs to be per-file, but I do think you need to say who owns the copyright somewhere in any source dist - it's not in your `LICENSE.txt` not your readme, etc (which the appendix at the end of the text encourages).

Reply:
Rustworkx licensing tracked in Qiskit/rustworkx#1270.